pallets/manual-migrations/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 Moonbeam foundation
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Multi Block Migration Pallet
+
+#![allow(non_camel_case_types)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+use frame_support::pallet;
+
+pub use pallet::*;
+
+#[pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	/// Pallet for multi block migrations
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	/// Configuration trait of this pallet.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// There are no more storage entries to be removed
+		AllStorageEntriesHaveBeenRemoved,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		// TODO(rodrigo): This extrinsic should be removed once LocalAssets pallet storage is removed
+		#[pallet::call_index(0)]
+		#[pallet::weight(
+			<T as frame_system::Config>::DbWeight::get()
+				.reads_writes((*limit + 1).into(), (*limit).into())
+		)]
+		pub fn clear_local_assets_storage(
+			origin: OriginFor<T>,
+			limit: u32,
+		) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+
+			let hashed_prefix = sp_io::hashing::twox_128("LocalAssets".as_bytes());
+
+			let keys_removed = match sp_io::storage::clear_prefix(&hashed_prefix, Some(limit)) {
```

Comment:
@RomarQ it's wrong and very unsafe here, the limit parameter for clear_prefix is the number of entries, not the proof_size.

You should figure out what is the maximum size of an entry for this pallet, and then there is 2 possible implementations:
1. Easier but not optimal, account a proof_size of LOCAL_ASSETS_MAXE_ENTRY_SIZE * number_of_keys
2. More optimal but more complex: read each entry one by one to get the lenght and refund `MAX_ENTRY_SIZE - actual_lenght` at each iteration

Reply:
Implemented the first approach